exchanges/huobi/huobi_wrapper.go

```diff
@@ -673,123 +976,258 @@ func (h *HUOBI) CancelAllOrders(orderCancellation *order.Cancel) (order.CancelAl
-			return cancelAllOrdersResponse, err
+		for i := range enabledPairs {
+			fpair, err := h.FormatExchangeCurrency(enabledPairs[i], asset.Spot)
+			if err != nil {
+				return cancelAllOrdersResponse, err
+			}
+			resp, err := h.CancelOpenOrdersBatch(orderCancellation.AccountID,
+				fpair.String())
+			if err != nil {
+				return cancelAllOrdersResponse, err
+			}
+			if resp.Data.FailedCount > 0 {
+				return cancelAllOrdersResponse,
+					fmt.Errorf("%v orders failed to cancel",
+						resp.Data.FailedCount)
+			}
+			if resp.Status == "error" {
+				return cancelAllOrdersResponse, errors.New(resp.ErrorMessage)
+			}
 		}
-
-		if resp.Data.FailedCount > 0 {
-			return cancelAllOrdersResponse,
-				fmt.Errorf("%v orders failed to cancel",
-					resp.Data.FailedCount)
+	case asset.CoinMarginedFutures:
+		if orderCancellation.Pair.IsEmpty() {
+			enabledPairs, err := h.GetEnabledPairs(asset.CoinMarginedFutures)
+			if err != nil {
+				return cancelAllOrdersResponse, err
+			}
+			for i := range enabledPairs {
+				fPair, err := h.FormatExchangeCurrency(enabledPairs[i], asset.CoinMarginedFutures)
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				a, err := h.CancelAllSwapOrders(fPair.String())
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				split := strings.Split(a.Successes, ",")
+				for x := range split {
+					cancelAllOrdersResponse.Status[split[x]] = "success"
+				}
+				for y := range a.Errors {
+					cancelAllOrdersResponse.Status[a.Errors[y].OrderID] = "fail"
+				}
+			}
+		} else {
+			fPair, err := h.FormatExchangeCurrency(orderCancellation.Pair, asset.CoinMarginedFutures)
+			if err != nil {
+				return cancelAllOrdersResponse, err
+			}
+			a, err := h.CancelAllSwapOrders(fPair.String())
+			if err != nil {
+				return cancelAllOrdersResponse, err
+			}
+			split := strings.Split(a.Successes, ",")
+			for x := range split {
+				cancelAllOrdersResponse.Status[split[x]] = "success"
+			}
+			for y := range a.Errors {
+				cancelAllOrdersResponse.Status[a.Errors[y].OrderID] = "fail"
+			}
 		}
-
-		if resp.Status == "error" {
-			return cancelAllOrdersResponse, errors.New(resp.ErrorMessage)
+	case asset.Futures:
+		if orderCancellation.Pair.IsEmpty() {
+			enabledPairs, err := h.GetEnabledPairs(asset.Futures)
+			if err != nil {
+				return cancelAllOrdersResponse, err
+			}
+			for i := range enabledPairs {
+				fPair, err := h.FormatExchangeCurrency(enabledPairs[i], asset.Futures)
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				a, err := h.FCancelAllOrders(fPair.Base.String(), fPair.String(), "")
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				split := strings.Split(a.Data.Successes, ",")
+				for x := range split {
+					cancelAllOrdersResponse.Status[split[x]] = "success"
+				}
+				for y := range a.Data.Errors {
+					cancelAllOrdersResponse.Status[strconv.FormatInt(a.Data.Errors[y].OrderID, 10)] = "fail"
```

Comment:
It looks like huobi actually returns error codes and reasons (err_code and err_msg). Might be helpful to include those here

Reply:
Bump

Reply:
retro bump